lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php

```diff
@@ -174,7 +183,13 @@ public function date($date = null, $locale = null, $useTimezone = true, $include
                     $timeType,
                     new \DateTimeZone($timezone)
                 );
+
+                //Know if the date contains time or not
+                if (!preg_match('/(.*).(\d{2}):(\d{2})/', $date)) {
+                    $date .= " 00:00:00";
```

Comment:
With $timeType set to \IntlDateFormatter::SHORT
`"00:00:00"` should be changed to "0:00am" to be parsed

http://php.net/manual/en/class.intldateformatter.php